packages/account-manifest/src/hooks/useGetFid.ts

```diff
@@ -0,0 +1,25 @@
+import { http } from 'viem';
+import { createPublicClient } from 'viem';
+import type { Address } from 'viem';
+import { optimism } from 'viem/chains';
+import { ID_REGISTRY_ABI } from '../constants';
+import { ID_REGISTRY_ADDRESS } from '../constants';
+
+export function useGetFid() {
+  return async function getFid(address: Address) {
+    const client = createPublicClient({
+      chain: optimism,
+      transport: http(),
+    });
+
+    // query the ID Registry contract for the fids custody address
+    const resolvedFid = await client.readContract({
+      address: ID_REGISTRY_ADDRESS,
+      abi: ID_REGISTRY_ABI,
+      functionName: 'idOf',
+      args: [address],
+    });
+
+    return Number(resolvedFid);
+  };
+}
```

Comment:
suggestion: We could simplify what the consumer has to do to use the value by adding the useState and useEffect into this hook itself.

E.g.

```tsx
const client = createPublicClient({
  chain: optimism,
  transport: http(),
});

export function useFid(address: Address) {
  const [fid, setFid] = useState<number | null>(null);

  useEffect(() => {
    async function getFid() {
      if (!address) return;

      const resolvedFid = await client.readContract({
        address: ID_REGISTRY_ADDRESS,
        abi: ID_REGISTRY_ABI,
        functionName: 'idOf',
        args: [address],
      });

      setFid(Number(resolvedFid));
    }

    getFid();
  }, [address]);

  return fid;
}
```